### ISSUE TITLE
semantic-release 15.9.x -> 15.10.x seem to be breaking this plugin

### ISSUE THREAD

**Issue by AVVS** [pmowrer/semantic-release-monorepo#65]
I'll try to take a look myself time permitting, but wanted to let you know.
Logs attached:

```
$ lerna exec --concurrency 1 -- semantic-release -e semantic-release-monorepo
lerna notice cli v3.4.3
lerna info versioning independent
lerna info Executing command in 1 package: "semantic-release -e semantic-release-monorepo"
[3:08:28 AM] [semantic-release] › ℹ  Running semantic-release version 15.10.3
[3:08:28 AM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/changelog"
[3:08:28 AM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[3:08:28 AM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/git"
[3:08:28 AM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[3:08:28 AM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[3:08:28 AM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[3:08:28 AM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[3:08:28 AM] [semantic-release] › ✔  Run automated release from branch master
[3:08:30 AM] [semantic-release] › ✔  Allowed to push to the Git repository
[3:08:31 AM] [semantic-release] › ℹ  Found git tag @microfleet/core-v13.0.2 associated with version 13.0.2
[3:08:31 AM] [semantic-release] › ℹ  Found 9 commits since last release
[3:08:31 AM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "[Function: ]"
[3:08:31 AM] [semantic-release] › ✖  Failed step "analyzeCommits" of plugin "[Function: ]"
[3:08:31 AM] [semantic-release] › ✖  An error occurred while running semantic-release: { TypeError: Expected `moduleId` to be of type `string`, got `object`
    at resolveFrom (/home/runner/core/node_modules/import-from/node_modules/resolve-from/index.js:11:9)
    at module.exports (/home/runner/core/node_modules/import-from/node_modules/resolve-from/index.js:34:41)
    at module.exports (/home/runner/core/node_modules/import-from/index.js:4:49)
    at requirePlugin (/home/runner/core/node_modules/semantic-release-plugin-decorators/src/index.js:4:33)
    at resolvePluginFn (/home/runner/core/node_modules/semantic-release-plugin-decorators/src/index.js:18:18)
    at /home/runner/core/node_modules/semantic-release-plugin-decorators/src/index.js:26:20
    at validator (/home/runner/core/node_modules/semantic-release/lib/plugins/normalize.js:30:28)
    at pReduce (/home/runner/core/node_modules/semantic-release/lib/plugins/pipeline.js:37:40)
    at Promise.all.then.value (/home/runner/core/node_modules/p-reduce/index.js:16:10)
    at process._tickCallback (internal/process/next_tick.js:68:7) pluginName: '[Function: ]' }
{ TypeError: Expected `moduleId` to be of type `string`, got `object`
    at resolveFrom (/home/runner/core/node_modules/import-from/node_modules/resolve-from/index.js:11:9)
    at module.exports (/home/runner/core/node_modules/import-from/node_modules/resolve-from/index.js:34:41)
    at module.exports (/home/runner/core/node_modules/import-from/index.js:4:49)
    at requirePlugin (/home/runner/core/node_modules/semantic-release-plugin-decorators/src/index.js:4:33)
    at resolvePluginFn (/home/runner/core/node_modules/semantic-release-plugin-decorators/src/index.js:18:18)
    at /home/runner/core/node_modules/semantic-release-plugin-decorators/src/index.js:26:20
    at validator (/home/runner/core/node_modules/semantic-release/lib/plugins/normalize.js:30:28)
    at pReduce (/home/runner/core/node_modules/semantic-release/lib/plugins/pipeline.js:37:40)
    at Promise.all.then.value (/home/runner/core/node_modules/p-reduce/index.js:16:10)
    at process._tickCallback (internal/process/next_tick.js:68:7) pluginName: '[Function: ]' }lerna ERR! semantic-release -e semantic-release-monorepo exited 1 in '@microfleet/core'
lerna ERR! semantic-release -e semantic-release-monorepo exited 1 in '@microfleet/core'
```

**Comment by pmowrer** [comment]
Oh boy, here we go again! Thanks for the report @AVVS!

**Comment by lifeiscontent** [comment]
I thought they were supposed to release semantically 💣  😅

**Comment by pmowrer** [comment]
Looks like the entire plugin format changed... I'll need to sit-down and work out what needs to be updated to support this.

Because `semantic-release` doesn't provide [a necessary plugin hook](https://github.com/semantic-release/semantic-release/pull/531) to make it work with a monorepo, this library has to monkey-patch `semantic-release` through its plugin system. Unfortunately, that makes this library very brittle...

**Comment by jimthedev** [comment]
Ouch. This one hurts. If there is anything I can do to help out please let me know.

**Comment by jimthedev** [comment]
So yeah it looks like this commit should have been a breaking change https://github.com/semantic-release/semantic-release/commit/5ba5010

**Comment by sanpoChew** [comment]
@pmowrer I'm pretty sure this fixes it: https://github.com/Updater/semantic-release-plugin-decorators/pull/9 we are using it fine now

**Comment by AnthonyNahas** [comment]
I am facing the same issue...

**Comment by pmowrer** [comment]
I posted a branch with what I believe will fix the issue here: https://github.com/pmowrer/semantic-release-monorepo/pull/80. Let know if it works for you

**Comment by AVVS** [comment]
moved to lerna since then, sadly don't have any monorepos to test it on

**Comment by pmowrer** [comment]
This should be fixed by in [v7.0.0](https://github.com/pmowrer/semantic-release-monorepo/releases/tag/v7.0.0).
